sklearn/utils/estimator_checks.py

```diff
@@ -2124,6 +2126,50 @@ def check_classifiers_one_label(name, classifier_orig):
         assert_array_equal(classifier.predict(X_test), y, err_msg=error_string_predict)
 
 
+@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+def check_classifiers_one_label_sample_weights(name, classifier_orig):
+    # check that classifiers accepting sample_weight fit or
+    # throws an ValueError with explicit message if
+    # the problem is reduce to one class.
+    error_fit = (
+        "{name} failed when fitted on one label after sample_weight "
+        "trimming. Error message is not explicit, it should have "
+        "'class'."
+    ).format(name=name)
+    error_predict = (
+        "{name} prediction results should only output the remaining class."
+    ).format(name=name)
+    classifier = clone(classifier_orig)
+    rnd = np.random.RandomState(0)
+    # X should be square for test on SVC with precomputed kernel.
+    X_train = rnd.uniform(size=(10, 10))
+    X_test = rnd.uniform(size=(10, 10))
+    y = np.arange(10) % 2
+    # keep only one class
+    sample_weight = y
+    # Test that fit won't raise an unexpected exception
+    if hasattr(classifier, "sample_weight"):
```

Comment:
codecov is complaining because `hasattr(classifier, "sample_weight")` is always false. Did you mean the following:

```suggestion
    if has_fit_parameter(estimator, "sample_weight"):
```